### PR TITLE
Vietnamese Translation

### PR DESCRIPTION
**Pull request by ShardNguyen**
This is the initialization of the Vietnamese translation of the game, might need some people to proof read it and give me feedback to help make it better

### CODE REVIEW

**Comment by MrZ626** [comment]
You can set this PR to "Draft" state to prevent merging

**Comment by Quochieu0211** [comment]
Touch weight should be "Khối lượng chạm" and origin weight should be "Khối lượng gốc"

**Review by SweetSea-ButImNotSweet — CHANGES_REQUESTED** [review]
Line 252:
``"Đây chỉ là một trò chơi xếp gạch *thông thường*. Không, thật đấy, chỉ có vậy thôi",``

It's a mistake that I accidentally made. It should be
``"Đây chỉ là một trò chơi xếp gạch *rất chi là thông thường*. Không hơn không kém"``

Well, but at least I knew this is my file with some Shard's edits. But anyway, I will pick his file and, continue to edit

**Comment by ShardNguyen** [comment]
> Touch weight should be "Khối lượng chạm" and origin weight should be "Khối lượng gốc"

Is there any alternative translation to this?

**Comment by Quochieu0211** [comment]
> > Touch weight should be "Khối lượng chạm" and origin weight should be "Khối lượng gốc"
> 
> 
> 
> Is there any alternative translation to this?

Touch weight: Cảm ứng trọng lượng
Origin weight: Trọng lượng gốc

**Comment by SweetSea-ButImNotSweet** [comment]
> > > Touch weight should be "Khối lượng chạm" and origin weight should be "Khối lượng gốc"
> > 
> > 
> > Is there any alternative translation to this?
> 
> Touch weight: Cảm ứng trọng lượng<br>Origin weight: Trọng lượng gốc

It's not a good translation, I think
How about if we explain it in Zictionary?

File: parts/language/lang_vi.lua [hunk 1]
```diff
@@ -883,7 +883,7 @@ return {
         ['blind_h']=        {"Invisible",         "KHÓ",         "Dành cho người đã có kinh nghiệm"},
         ['blind_l']=        {"Invisible",         "KHÓ+",        "Dành cho người chơi chuyên nghiệp"},
         ['blind_u']=        {"Invisible",         "?",              "Bạn đã sẵn sàng chưa?"},
-        ['blind_wtf']=      {"Invisible",         "WTF",            "Bạn chưa chuẩn bị cho màn này đâu!"},
+        ['blind_wtf']=      {"Invisible",         "CLGT?",            "Bạn chưa chuẩn bị cho màn này đâu!"},
```
Comment (SweetSea-ButImNotSweet) [inline parts/language/lang_vi.lua:886]: I think we can use "**C**ái **Q**uái **G**ì **Đ**ây **?**" may be better<br>
But you can keep your change

File: parts/language/lang_vi.lua [hunk 2]
```diff
@@ -990,7 +990,7 @@ return {
         "It’s possible to finish 40L without rotation buttons.",
         "Join our Discord!",
         "l-=-1",
-        "Let the bass kick!",
+        "Biết ông Thương không?",
         "Low frame rates reduce your gaming experience.",
         "LrL RlR LLr RRl RRR LLL FFF RfR RRf rFF",
```
Comment (SweetSea-ButImNotSweet) [inline parts/language/lang_vi.lua:995]: This is a guide to solve a rubik. If I remember correctly, it's name is "||Sexy|| dance". (I'm not joking)

**Comment by SweetSea-ButImNotSweet** [comment]
~~``Số lần tạm dừng`` or ``Thời gian tạm dừng`` is better?~~
I just thought about it again and realized I was wrong too. I will fix it now
image

Here is where that text will appear
image

By the way, I also had to fork this repo to update easily because Shard had forked before I did

File: parts/language/lang_vi.lua (hunk 1 above)
Comment (ShardNguyen) [reply parts/language/lang_vi.lua:886]: I think CLGT? is better since it has the same vibe as WTF?

Comment (ShardNguyen) [reply parts/language/lang_vi.lua:886]: "CLGT" has the same literal style as "WTF" while also containing a swear word in there as well

**Comment by ShardNguyen** [comment]
> By the way, I also had to fork this repo to update easily because Shard had forked before I did Here is it, if you want to have a look: https://github.com/SweetSea-ButImNotSweet/Techmino

I think the main purpose of this counter is to count how many pauses has been occurred so "Số lần" is better. The timer is just the extras

File: parts/language/lang_vi.lua (hunk 1 above)
Comment (Quochieu0211) [reply parts/language/lang_vi.lua]: I could also came with "Cái Đ** Gì Đây?" since it was a common swear word used by Vietnamese teenagers

**Comment by SweetSea-ButImNotSweet** [comment]
Shard, something I want to notice
Can you clear the comment at the line 952 in your translation? **VERY IMPORTANT**

**Comment by SweetSea-ButImNotSweet** [comment]
Ok, you maybe don't notice
So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?

**Comment by ShardNguyen** [comment]
> Ok, you maybe don't notice
> So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?

For me, I will keep it the same since it is a terminology

**Comment by ShardNguyen** [comment]
> Ok, you maybe don't notice
> So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?

Hmm, that actually sounds pretty good to me

**Comment by SweetSea-ButImNotSweet** [comment]
> > Ok, you maybe don't notice
> > So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?
> 
> Hmm, that actually sounds pretty good to me

So, should we use this for replacing ``Gạch ma``?

**Comment by ShardNguyen** [comment]
> > > Ok, you maybe don't notice
> > > So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?
> > 
> > Hmm, that actually sounds pretty good to me
> 
> So, should we use this for replacing ``Gạch ma``?

Yeah, sounds good to me

**Comment by Quochieu0211** [comment]
> > > > Ok, you maybe don't notice
> > > > So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?
> > > 
> > > 
> > > Hmm, that actually sounds pretty good to me
> > 
> > 
> > So, should we use this for replacing `Gạch ma`?
> 
> Yeah, sounds good to me

I don't think we call that a gạch ma because that would makes absolutely no sense, khối ma would be a better word for this

**Comment by SweetSea-ButImNotSweet** [comment]
Shard, I notice you just changed ``Xoas`` to ``Dọn``, well sound better
I will apply it into my translation

**Comment by SweetSea-ButImNotSweet** [comment]
> > > > > Ok, you maybe don't notice
> > > > > So iirc, in Gameloft TETRIS, the Ghost option is translated to "Bóng gạch". Any idea?
> > > > 
> > > > 
> > > > Hmm, that actually sounds pretty good to me
> > > 
> > > 
> > > So, should we use this for replacing `Gạch ma`?
> > 
> > 
> > Yeah, sounds good to me
> 
> I don't think we call that a gạch ma because that would makes absolutely no sense, khối ma would be a better word for this

Yes, but that only if I can use the word "Khối"
Also when I was commenting here, Shard had changed the word "Gạch ma" to "Bóng gạch"

**Comment by ShardNguyen** [comment]
> I don't think we call that a gạch ma because that would makes absolutely no sense, khối ma would be a better word for this

I think "Bóng gạch" should be the best for this one, which is what Sweet has suggested earlier

"Bóng gạch" can be understood as the reflection of a block, which is what 'Ghost Pieces' does

File: parts/language/lang_vi.lua [hunk 3]
```diff
@@ -0,0 +1,1070 @@
...
+    joinRoom="$1 vừa vào phòng.",
+    leaveRoom="$1 vừa rời phòng.",
+    roomRemoved="Phòng đã bị loại bỏ",
+    ready="Sẵn sàng",
+    spectating="Đang theo dõi",
+
+
+
+    keySettingInstruction="Nhấn một phím để chỉ định phím đó\nescape (esc): Huỷ chỉ định\nbackspace: Xoá phím đã chỉ định",
+    customBGhelp="Chèn một tấm ảnh vào đây để áp dụng ảnh nền tuỳ chỉnh",
```
Comment (SweetSea-ButImNotSweet) [inline parts/language/lang_vi.lua:196]: Line 196: I think using ``Chèn`` (insert) can make some players misunderstanding
How about if we use ``Thả`` or ``Ném``?

**Comment by SweetSea-ButImNotSweet** [comment]
Thanks
This translation is ready to be merged
(Hope from the time I commented this until it is merged there will be no unexpected events)

**Comment by MrZ626** [comment]
Missing "," in file `lang.lua` (line 12), and we should use space as indent (not tab)

**Comment by SweetSea-ButImNotSweet** [comment]
> Missing "," in file `lang.lua (end of line 11)`, and we should use space as indent (not tab)

oh, okay. I will edit
Anyway, I tried to call Shard but seem like I think I need to copy the Shard's one and fix. (I can't edit directly so I will send the link instead)

**Comment by SweetSea-ButImNotSweet** [comment]
> > Missing "," in file `lang.lua (end of line 11)`, and we should use space as indent (not tab)
> 
> oh, okay. I will edit Anyway, I tried to call Shard but seem like I think I need to copy the Shard's one and fix. (I can't edit directly so I will send the link instead)

Here is it, I don't have any idea then:
https://github.com/SweetSea-ButImNotSweet/Techmino/blob/VietnameseTranslation/parts/language/lang_vi.lua
https://github.com/SweetSea-ButImNotSweet/Techmino/blob/VietnameseTranslation/parts/scenes/lang.lua

File: main.lua [hunk 4]
```diff
@@ -502,7 +502,7 @@ LANG.init('zh',
         ja=require'parts.language.lang_ja',
         symbol=require'parts.language.lang_symbol',
         zh_code=require'parts.language.lang_zh_code',
-		vi=require'parts.language.lang_vi'
+vi=require'parts.language.lang_vi'
```
Comment (MrZ626) [inline main.lua:505]: missing spaces

**Review by Neubulae — APPROVED** [review]

**Review by MrZ626 — APPROVED** [review]

**Pull request merged**